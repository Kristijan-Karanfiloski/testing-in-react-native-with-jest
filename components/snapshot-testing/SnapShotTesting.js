import { Text, View } from "react-native";

const SnapShotTesting = () => {
  return (
    <>
      <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
        <Text style={{ fontSize: 23 }}>Snapshot Testing</Text>
      </View>
    </>
  );
};

export default SnapShotTesting;
