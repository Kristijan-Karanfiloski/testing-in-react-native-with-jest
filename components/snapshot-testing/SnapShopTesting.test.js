import renderer from "react-test-renderer";
import SnapShotTesting from "./SnapShotTesting";

describe("Snapshot Testing", () => {
  it("should make a snap shot of the component", () => {
    const snapshot = renderer.create(<SnapShotTesting />).toJSON();

    expect(snapshot).toMatchSnapshot();
  });

  it("should check the matchers and fail", () => {
    const user = {
      createdAt: new Date(),
      id: Math.floor(Math.random() * 20),
      name: "LeBron James",
    };

    expect(user).toMatchSnapshot();
  });
  it("should check the matchers and pass", () => {
    const user = {
      createdAt: new Date(),
      id: Math.floor(Math.random() * 20),
      name: "LeBron James",
    };

    expect(user).toMatchSnapshot({
      createdAt: expect.any(Date),
      id: expect.any(Number),
      name: expect.any(String),
    });
  });
});
