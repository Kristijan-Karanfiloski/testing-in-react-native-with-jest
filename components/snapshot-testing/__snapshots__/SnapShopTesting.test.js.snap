// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing should check the matchers and fail 1`] = `
{
  "createdAt": 2023-11-15T14:03:08.489Z,
  "id": 15,
  "name": "LeBron James",
}
`;

exports[`Snapshot Testing should check the matchers and pass 1`] = `
{
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": Any<String>,
}
`;

exports[`Snapshot Testing should make a snap shot of the component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 23,
      }
    }
  >
    Snapshot Testing
  </Text>
</View>
`;
